### PR TITLE
Complete code rewrite

### PR DESCRIPTION
**Pull request by paranerd**
This is v2. Main changes:
- TypeScript
- Tests in the repo
- Code bundling before push to Apps Script
- Create and update Audience Lists in a single sheet
- Edit basically anything about the Audience List that you can create
- Put all menus and controls into the sidebar

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,36 @@
...
-1.  Create an *Apps Script* project that is linked to this spreadsheet by
-    navigating to *Tools > Script Editor*. Alternatively open the script
-    directly from the
-    [Apps Script Project Dashboard](https://script.google.com/home/all).
-1.  Copy the code samples provided in [resources/](resources) to the script
-    project created above. Consider using
-    [clasp](https://github.com/google/clasp) for easier code management.
-1.  Follow the instructions in the *Usage Guide* tab of the spreadsheet to start
-    using the solution
+<!--
```
Comment (mohabfekry) [inline README.md:1]: We don't need a specific license header for README.md files

Comment (paranerd) [reply README.md:1]: Shall I remove it or can it stay in there because no harm done?

File: src/facade/cm360.ts [hunk 2]
```diff
@@ -25,41 +28,49 @@
...
+    accountData: { networkId: string; advertiserId: string },
+    apiFirst: boolean,
+    advertisersFilter: string[] = [],
+    campaignManagerServiceWrapper?: GoogleAppsScript.CampaignManager.CampaignManager
+  ) {
     this.accountData_ = accountData;
-
-    /** @private @const {boolean} */
     this.apiFirst_ = apiFirst;
+    this.advertisersFilter = advertisersFilter || [];
```
Comment (mohabfekry) [inline src/facade/cm360.ts]: the || [] is unnecessary as this is already added in the constructor param

File: src/static/audience.html [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      this.lifeSpan_ = lifeSpan;
+
+      /** @private @const {string|undefined} */
+      this.floodlightId_ = String(floodlightId);
+
+      /** @private @const {string|undefined} */
+      this.floodlightName_ = floodlightName;
+
+      /** @private {!Array<!AudienceRule>} */
+      this.rules_ = rules || [];
```
Comment (mohabfekry) [inline src/static/audience.html]: move the default param ( || [] ) to the constructor

File: src/static/audience.html [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+      /** @private {!Array<!string>} */
+      this.shares_ = shares || [];
```
Comment (mohabfekry) [inline src/static/audience.html]: move the default param ( || [] ) to the constructor

**Review by mohabfekry — COMMENTED** [review]
Thanks looks great! Added a few minor comments

**Review by mohabfekry — DISMISSED** [review]

**Review by mohabfekry — APPROVED** [review]

**Pull request merged**